src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -581,9 +586,7 @@ private static void ReverseInner<T>(ref T elements, nuint length)
             ref T last = ref Unsafe.Subtract(ref Unsafe.Add(ref first, (int)length), 1);
             do
             {
-                T temp = first;
-                first = last;
-                last = temp;
+                (last, first) = (first, last);
```

Comment:
This is regression for large structs: https://github.com/dotnet/runtime/pull/64412#discussion_r864742976

Reply:
We should really fix that, either in the JIT or special-cased in Roslyn or both. VS analyzers/fixers push developers to use this pattern now for swaps, such that it's becoming a commonly-used idiom.

Reply:
Should that fixer be disabled for now, for structs at least, and only enabled once this perf bug is fixed? It does not sound right for the fixers to suggest performance regressions.

> We should really fix that, either in the JIT or special-cased in Roslyn or both. 

This should better be fixed in Roslyn. Fixing it in the JIT would mean that the fix can be only suggested for new runtimes that have the fix.

Reply:
I don't know that disabling the rule really helps. The tuple deconstruction feature makes this a natural thing to write, and it's advocated for in many other places; the rule is largely just reflecting that. Better for https://github.com/dotnet/roslyn/issues/53300 to just be fixed, IMO.
cc: @jaredpar

Reply:
The fixer is appropriate. It's producing an idiomatic pattern. Given that we explicitly designed this at the language level to be a way to swap, the expectation is that the compilers will then implement that well.

Reply:
Thanks, I wasn't aware of this. I reverted it for now

Reply:
Will take a look at this post C# 11. That part of tuples went through a lot of back and forth about "do we need to compile down to tuples or can we push through and just look at the begin / end variables". Largely we've come down on "just push through" but still cases we missed around what can be optimized.